sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -1274,7 +1277,7 @@ private Mono<Boolean> checkIMDSAvailable(String endpoint) {
             try {
                 connection = (HttpURLConnection) url.openConnection();
                 connection.setRequestMethod("GET");
-                connection.setConnectTimeout(500);
+                connection.setConnectTimeout(1000);
```

Comment:
Is this something that should be configurable? That way we can have a default and an escape hatch if needed.

There are also other locations where we're using `HttpURLConnection` in this class, should those also have a connection timeout?

Reply:
nope, this is just to align across languages and is intended to be at this stage only, and risk is low as we're only increasing it, with users not facing any issue at lower value too.

Doesn't need to be configurable,